apparmor-mode.el

```diff
@@ -126,7 +126,7 @@
 
 (defvar apparmor-mode-profile-name-regexp "[[:alnum:]]+")
 
-(defvar apparmor-mode-profile-attachment-regexp "[[:alnum:]*@/_{},-]+")
+(defvar apparmor-mode-profile-attachment-regexp "[][[:alnum:]*@/_{},-.?]+")
```

Comment:
I think instead these square brackets should be included in the existing group - ie.

``` emacs-lisp
(defvar apparmor-mode-profile-attachment-regexp "[[:alnum:]*@/_{}\[\],-.?]+")
```

Reply:
No. It will fail. Please take a look of GNU official documentation about this:

[Regexps (GNU Emacs Manual)](https://www.gnu.org/software/emacs/manual/html_node/emacs/Regexps.html)

> To include a ‘\]’ in a character set, you must make it the first character. For example, ‘\[\]a\]’ matches ‘\]’ or ‘a’. To include a ‘\-’, write ‘\-’ as the last character of the set, tho you can also put it first or after a range. Thus, ‘\[\]-\]’ matches both ‘\]’ and ‘\-’.


![Screenshot_20240117_220558](https://github.com/alexmurray/apparmor-mode/assets/1370070/d5b9562f-7539-40e2-89f1-ce0a1ea3c486)
![Screenshot_20240117_220605](https://github.com/alexmurray/apparmor-mode/assets/1370070/ff8e7741-2aa7-420b-837f-af1dafcad920)

Reply:
ah ok - thanks for the detailed response.